File: Contact.Api/ApiEndpoints.cs
namespace Contact.Api;

public class ApiEndpoints
{
    private const string ApiBase = "api";

    public static class Contact
    {
        private const string Base = $"{ApiBase}/contacts";

        public const string Create = Base;

        public const string Get = $"{Base}/{{id:guid}}";

        public const string GetAll = Base;

        public const string Update = $"{Base}/{{id:guid}}";

        public const string Delete = $"{Base}/{{id:guid}}";
    }
}

File: contact.client/src/api/ApiEndpoints.js
const API_BASE_URL = process.env.REACT_APP_API_BASE_URL; // Base api URL is here

export const ApiEndpoints = {
    Contact: {
        Create: `${API_BASE_URL}/contacts`,
        Get: (id) => `${API_BASE_URL}/contacts/${id}`,
        GetAll: (page, pagesize) => `${API_BASE_URL}/contacts?pagesize=${pagesize}&page=${page}`,
        Update: (id) => `${API_BASE_URL}/contacts/${id}`,
        Delete: (id) => `${API_BASE_URL}/contacts/${id}`,
    }
};

export default ApiEndpoints;